core/src/test/java/kafka/test/api/ShareConsumerTest.java

```diff
@@ -1806,6 +1816,181 @@ public void testShareAutoOffsetResetByDurationInvalidFormat() throws Exception {
+                .partitions().stream()
+                .filter(info -> info.partition() == shareGroupStateTp)
+                .map(info -> info.leader().id())
+                .toList();
+
+            assertEquals(1, curShareCoordNodeId.size());
+
+            // shutdown the coordinator
+            cluster.shutdownBroker(curShareCoordNodeId.get(0));
+
+            // give some breathing time
+            TimeUnit.SECONDS.sleep(2L);
+
+            List<Integer> newShareCoordNodeId = admin.describeTopics(List.of(Topic.SHARE_GROUP_STATE_TOPIC_NAME)).topicNameValues().get(Topic.SHARE_GROUP_STATE_TOPIC_NAME).get()
+                .partitions().stream()
+                .filter(info -> info.partition() == shareGroupStateTp)
+                .map(info -> info.leader().id())
+                .toList();
+
+            assertEquals(1, newShareCoordNodeId.size());
+            assertNotEquals(curShareCoordNodeId.get(0), newShareCoordNodeId.get(0));
+
+            // again produce to same topic partition
+            try (Producer<byte[], byte[]> producer = createProducer()) {
+                ProducerRecord<byte[], byte[]> record = new ProducerRecord<>(
+                    tpMulti.topic(),
+                    tpMulti.partition(),
+                    null,
+                    "key".getBytes(),
+                    "value".getBytes()
+                );
+                IntStream.range(0, 10).forEach(__ -> producer.send(record));
+                producer.flush();
+            }
+
+            // consume messages should only be possible if partition and share coord has moved
+            // from shutdown broker since we are only producing to partition 0 of topic.
+            try (ShareConsumer<byte[], byte[]> shareConsumer = createShareConsumer(groupId)) {
+                shareConsumer.subscribe(List.of(topicName));
+                alterShareAutoOffsetReset(groupId, "earliest");
+                ConsumerRecords<byte[], byte[]> records = shareConsumer.poll(Duration.ofMillis(5000));
+                assertEquals(20, records.count());
+            }
+
+            verifyShareGroupStateTopicRecordsProduced();
+        }
+    }
+
+    @ClusterTest(
+        brokers = 3,
+        serverProperties = {
+            @ClusterConfigProperty(key = "auto.create.topics.enable", value = "false"),
+            @ClusterConfigProperty(key = "group.coordinator.rebalance.protocols", value = "classic,consumer,share"),
+            @ClusterConfigProperty(key = "group.share.enable", value = "true"),
+            @ClusterConfigProperty(key = "group.share.partition.max.record.locks", value = "10000"),
+            @ClusterConfigProperty(key = "group.share.record.lock.duration.ms", value = "15000"),
+            @ClusterConfigProperty(key = "offsets.topic.num.partitions", value = "3"),
+            @ClusterConfigProperty(key = "offsets.topic.replication.factor", value = "3"),
+            @ClusterConfigProperty(key = "share.coordinator.state.topic.min.isr", value = "1"),
```

Comment:
Same comments as previous test regarding min.isr and replication factor defaults which are appropriate without overrides.